### ISSUE TITLE
Cancelling the login dialog shouldn't show a "check credentials" error

### ISSUE THREAD

**Issue by jgat** [CSSE1001/MyPyTutor#63]
When prompted to log in, if the user clicks "cancel", they are shown an error message asking them to check their credentials. Proper behaviour would be to show them no message.

Optionally, show a short message (something like "Some functionality, such as submitting answers, requires you to be logged in. Log in using your UQ credentials.") on the login prompt.

**Comment by sapi** [comment]
From memory this is because the sso code currently doesn't distinguish between failing to login and cancelling the login action.

We'll have to fix that so that we can get rid of the redundant message while still informing the user of genuine failures.

**Comment by jgat** [comment]
Yeah, the `SessionManager.login` method needs something to change after the call to `LoginDialog`.

Ideally, `WebAPI.login` should have three possible outcomes - (1) if the user is logged in, (2) if the login attempt failed, (3) if the attempt was cancelled.

**Comment by sapi** [comment]
Yeah.

My vote is probably for (1) and (3) to be booleans, and (2) to be the existing `AuthError` behaviour.  (On the `SessionManager` side, at least.  `WebAPI` should arguably wrap the latter in one of its own exceptions.)
